XMonad/Hooks/EwmhDesktops.hs

```diff
@@ -171,10 +179,17 @@ ewmhDesktopsLogHookCustom t = withWindowSet $ \s -> do
         setNumberOfDesktops (length desktopNames)
         setDesktopNames desktopNames
 
-    -- Set client list; all windows, with focused windows last
-    let clientList = nub . concatMap (maybe [] (\(W.Stack x l r) -> reverse l ++ r ++ [x]) . W.stack) $ ws
+    -- Set client list which should be sorted by window age. We just
+    -- guess that StackSet contains windows list in this order which
+    -- isn't true but at least gives consistency with windows cycling
+    let clientList = nub . concatMap (reverse . W.integrate' . W.stack) $ ws
```

Comment:
Why is there a `reverse` here?

Reply:
If we don't have reverse here taffybar will display windows icons in a reversed order, but I'm not sure why. I'm fine if we remove reverse here because ordering can be easily adjusted on taffybar side,

Reply:
Removed reverse as it seems this way it'll be closer to specs.